yucca/network_architectures/networks/YuccaNet.py

```diff
@@ -30,14 +30,16 @@ def load_state_dict(self, target_state_dict, *args, **kwargs):
         }
         super().load_state_dict(target_state_dict, *args, **kwargs)
 
-    def predict(self, mode, data, patch_size, overlap, mirror=False):
+    def predict(self, mode, data, patch_size, overlap, sliding_window_prediction=True, mirror=False):
         if torch.cuda.is_available():
             data = data.to("cuda")
 
-        if mode == "3D":
-            predict = self._predict3D
-        if mode == "2D":
-            predict = self._predict2D
+        if not sliding_window_prediction:
+            return self._predict(data)
+        elif mode == "3D":
+            predict = self._sliding_window_predict3D
+        elif mode == "2D":
+            predict = self._sliding_window_predict2D
```

Comment:
While python allows for otherwise, i find it best that a branch always returns values of the same type. The first branch returns a value, while the two other branches returns methods. I would prefer if we rewrote it to:
```python
if not sliding_window_prediction:
    return self._predict(data)
            
if mode == "3D":
    predict = self._sliding_window_predict3D
elif mode == "2D":
     predict = self._sliding_window_predict2D
 ...
```